cmake_modules/CheckAtomicIntrinsic.cmake

```diff
@@ -111,13 +104,11 @@ if( SUPPORT_C11 AND PARSEC_ATOMIC_USE_C11_ATOMICS )
       cmake_push_check_state()
       list(APPEND CMAKE_REQUIRED_LIBRARIES atomic)
       unset( PARSEC_ATOMIC_USE_C11_128 CACHE )
-      CHECK_C_SOURCE_COMPILES("
+      check_c_source_runs("
           #include <stdatomic.h>
           int main(void) {
-              __int128_t where = 0, expected = 0;
-              if( !atomic_compare_exchange_strong( (_Atomic __int128_t*)&where, &expected, 1 ) )
-                  return -1;
-              return 0;
+            __int128_t where = 0;
+            return !atomic_is_lock_free(&where);
```

Comment:
Why is the result negated here? Should we assert on `atomic_is_lock_free`?

Reply:
because cmake's `check_c_source_runs` looks at the successful execution of the code, which means we need to return 0 if everything works as expected. On the other hand, `atomic_is_lock_free` returns true if the resulting operation is truly atomic.